UnityProject/Assets/Scripts/Objects/Enterable.cs

```diff
@@ -6,11 +6,41 @@
 namespace Objects
 {
 	/// <summary>
-	/// Allows to trigger an event when a player enter in the space occupied by this component.
+	/// Allows to trigger an event when a player enter the tile of this object.
+	/// Does trigger for ghosts, so check in OnStep
+	/// ServerSide Only
 	/// </summary>
-	public interface IEnterable
+	public interface IPlayerEntersTile
 	{
-		public abstract void OnStep(GameObject eventData);
-		public abstract bool WillStep(GameObject eventData);
+		/// <summary>
+		/// Check whether the player should be allowed to trigger the OnPlayerStep logic
+		/// </summary>
+		/// <param name="playerScript">Playerscript of the player stepping</param>
+		/// <returns>False to block OnPlayerStep, true to run OnPlayerStep</returns>
+		public abstract bool CanPlayerStep(PlayerScript playerScript);
+
+		/// <summary>
+		/// Action when player steps on tile
+		/// </summary>
+		/// <param name="playerScript">Playerscript of the player stepping</param>
+		public abstract void OnPlayerStep(PlayerScript playerScript);
+	}
+
+	/// <summary>
+	/// Allows to trigger an event when a object (includes mobs, objects and items) enter the tile of this object.
+	/// ServerSide Only
+	/// </summary>
+	public interface IObjectEntersTile
+	{
+		/// <summary>
+		/// Check whether this mobs, object or item should be allowed to trigger the OnObjectEnter logic
+		/// </summary>
+		/// <returns>False to block OnObjectEnter, true to run OnObjectEnter</returns>
+		public abstract bool CanObjectEnter(GameObject eventData);
```

Comment:
similarly, ``CanObjectEnter`` sounds like it will prevent the object from entering the tile which isn't the case. ``WillAffectObject`` might be clearer(?)